File: dp/canSum.js
let canSum = (target, list, memo = {}) => {
  //if (target in memo) return memo[target];
  if (target === 0) return true;
  if (target < 0) return false;

  for (let val of list) {
    const rem = target - val;
    if (canSum(rem, list, memo)) {
      //memo[target] = true;
      return true;
    }
  }
  //memo[target] = false;
  return false;
};

console.log(canSum(300, [7, 14]));

File: dp/tabulation/can_sum.py
def can_sum(target, arrays):
    dp = [False] * (target + 1)

    dp[0] = True
    for i in range(0, len(dp)):
        if dp[i]:
            for j in arrays:
                if i + j <= target:
                    dp[i + j] = True
        print(dp)
    return dp[target]


print(can_sum(300, [7, 14]))
